yarn-project/scripts/get_dependencies.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -eu
+
+REPO=$1
+
+cd "$(dirname "$0")"
+
+echo yarn-project-base
```

Comment:
Meh. Work for works sake. With chatgpt this took me 5 minutes and 1 line of jq. Glorious!

Reply:
But to be less dismissive, and for those reading up on this PR later, the downsides of this kind of approach is that is can slow down a bit if you go to far with processes calling processes calling processes...
I ran into that earlier with some of the dependency calculation stuff, but resolved it with an assoc array and early outs... an algorithm a sensible person would have done in another language anyway.
But yeah, if at some point this becomes unfit for purpose then can reconsider.

If this kind of code is "ballooning", my thought would be why are we adding so much devops stuff!? Rather than do we need to start changing how we do devops stuff.

Less is more, KISS, etc etc

Reply:
I woke up with a bit more to say on this.

First r.e. performance, it already takes ~3s to compute the rebuild patterns for e.g. end-to-end, which has a larger graph. If this were to double or more, one might question doing it in e.g. node.

However, then you need to install node in our tiny build image, so we need to balance that tradeoff in additional download time, or installing nodejs in the alpine container at runtime. A quick test on mainframe makes the later seem near instant, but mainframe has super fast stable network (cci questionable).

Installing nodejs grows the uncompressed image from 188MB to 239MB, however the existing build image is only 53MB compressed, so extending that compression ratio we could expect the image data download to increase to 67MB. This is perhaps negligible, but it's also just a quick estimate.